contracts/src/TokenSpoke/TokenSpoke.sol

```diff
@@ -16,60 +15,55 @@ import {
     SingleHopCallMessage,
     MultiHopSendMessage,
     MultiHopCallMessage,
-    RegisterDestinationMessage
-} from "./interfaces/ITeleporterTokenBridge.sol";
-import {
-    ITeleporterTokenDestination,
-    TeleporterTokenDestinationSettings
-} from "./interfaces/ITeleporterTokenDestination.sol";
-import {SendReentrancyGuard} from "./utils/SendReentrancyGuard.sol";
-import {TokenScalingUtils} from "./utils/TokenScalingUtils.sol";
+    RegisterSpokeMessage
+} from "../interfaces/ITokenBridge.sol";
+import {TeleporterMessageInput, TeleporterFeeInfo} from "@teleporter/ITeleporterMessenger.sol";
+import {TeleporterOwnerUpgradeable} from "@teleporter/upgrades/TeleporterOwnerUpgradeable.sol";
 import {IWarpMessenger} from
     "@avalabs/subnet-evm-contracts@1.2.0/contracts/interfaces/IWarpMessenger.sol";
 import {SafeERC20TransferFrom} from "@teleporter/SafeERC20TransferFrom.sol";
 import {IERC20} from "@openzeppelin/contracts@4.8.1/token/ERC20/ERC20.sol";
+import {SendReentrancyGuard} from "../utils/SendReentrancyGuard.sol";
+import {TokenScalingUtils} from "../utils/TokenScalingUtils.sol";
 
 /**
  * THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
  * DO NOT USE THIS CODE IN PRODUCTION.
  */
 
 /**
- * @title TeleporterTokenDestination
- * @dev Abstract contract for a Teleporter token bridge that receives tokens from a {TeleporterTokenSource} in exchange for the tokens of this token bridge instance.
+ * @title TokenSpoke
+ * @dev Abstract contract for a token bridge spoke that receives tokens from its specified token hub instance, and
+ * allows for burning that token to redeem the backing asset on the hub chain, or bridging to other spokes.
  *
  * @custom:security-contact https://github.com/ava-labs/teleporter-token-bridge/blob/main/SECURITY.md
  */
-abstract contract TeleporterTokenDestination is
-    ITeleporterTokenDestination,
-    TeleporterOwnerUpgradeable,
-    SendReentrancyGuard
-{
+abstract contract TokenSpoke is ITokenSpoke, TeleporterOwnerUpgradeable, SendReentrancyGuard {
```

Comment:
imo matters most when you're deriving off contracts that also derive off the same contract, like a diamond structure. https://github.com/ava-labs/subnet-evm/blob/master/precompile/contracts/warp/contract_warp_handler.go#L72 has some docs on it

Reply:
The order here does "matter" in determining the order of inheritance. In a hypothetical child contract that inherits from multiple parents that have conflicting function names, the child contract will use the implementation of the last parent listed in the inheritance list. The solidity docs for this are here: https://docs.soliditylang.org/en/develop/contracts.html#multiple-inheritance-and-linearization

In the case of `SendReentrancyGuard`, I don't think it makes a difference since the 
`sendNonReentrant` method is overridden anywhere, but I'll move it to be last in the inheritance list anywhere it is used for consistency.